packages/app/src/cli/services/context.ts

```diff
@@ -149,47 +149,53 @@ export async function ensureDevContext(options: DevContextOptions, token: string
     return buildOutput(selectedApp, selectedStore, cachedInfo)
   }
 
-  const organization = await fetchOrgFromId(orgId, token)
-  if (!organization) throw new BugError(`Couldn't find Organization with id ${orgId}.`)
-
-  if (!selectedApp) {
-    if (cachedInfo?.appId) {
-      const app = await fetchAppFromApiKey(cachedInfo.appId, token)
-      if (!app) throw new BugError(`Couldn't find App with apiKey ${cachedInfo.appId}.`)
-      selectedApp = app
-    } else {
-      const {apps} = await fetchOrgAndApps(orgId, token)
-      const localAppName = await loadAppName(options.directory)
-      selectedApp = await selectOrCreateApp(localAppName, apps, organization, token)
-    }
-  }
-
-  setAppInfo({
-    appId: selectedApp.apiKey,
-    title: selectedApp.title,
-    directory: options.directory,
-    orgId,
-  })
-
-  if (!selectedStore) {
-    if (cachedInfo?.storeFqdn) {
-      const result = await fetchStoreByDomain(organization.id, token, cachedInfo.storeFqdn)
-      if (result?.store) {
-        await convertToTestStoreIfNeeded(result.store, organization, token)
-        selectedStore = result.store
-      } else {
-        throw new BugError(`Couldn't find Store with domain ${cachedInfo.storeFqdn}.`)
+  const [organization, _selectedApp, _selectedStore] = await Promise.all([
+    (async (): Promise<Organization> => {
+      const organization = await fetchOrgFromId(orgId, token)
+      if (!organization) throw new BugError(`Couldn't find Organization with id ${orgId}.`)
+      return organization
+    })(),
+    (async (): Promise<OrganizationApp | undefined> => {
+      if (selectedApp) return selectedApp
+      if (cachedInfo?.appId) {
+        const app = await fetchAppFromApiKey(cachedInfo.appId, token)
+        if (!app) throw new BugError(`Couldn't find App with apiKey ${cachedInfo.appId}.`)
+        return app
       }
-    } else {
-      const allStores = await fetchAllDevStores(orgId, token)
-      selectedStore = await selectStore(allStores, organization, token)
-    }
+    })(),
+    (async (): Promise<OrganizationStore | undefined> => {
+      if (selectedStore) return selectedStore
+      if (cachedInfo?.storeFqdn) {
+        const result = await fetchStoreByDomain(orgId, token, cachedInfo.storeFqdn)
+        if (result?.store) {
+          await convertToTestStoreIfNeeded(result.store, orgId, token)
+          return result.store
+        } else {
+          throw new BugError(`Couldn't find Store with domain ${cachedInfo.storeFqdn}.`)
+        }
+      }
```

Comment:
could you extract each promise code into its own function? 
With the new changes (which I think make sense) it's a bit more difficult to read the code

Reply:
Agreed. It'd have a positive impact on the readability.